modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/http/HTTPErrorHandler.java

```diff
@@ -38,11 +38,14 @@
         service = ServerConnectorErrorHandler.class)
 public class HTTPErrorHandler implements ServerConnectorErrorHandler {
 
-    private static final Logger log = LoggerFactory.getLogger(HTTPErrorHandler.class);
-
     @Override
-    public void handleError(Exception e, CarbonMessage carbonMessage, CarbonCallback callback) {
-        callback.done(createErrorMessage(e.getMessage(), 500));
+    public void handleError(Exception exception, CarbonMessage carbonMessage, CarbonCallback callback) {
+        if (exception.getCause() instanceof ServiceNotFoundException ||
```

Comment:
This looks good! Wouldn't it also be better to check for the exception object similar to the exception cause as this method can be invoked with a ServiceNotFoundException or ResourceNotFoundException directly?